mungegithub/mungers/lgtm_handler.go

```diff
@@ -81,39 +81,51 @@ func (h *LGTMHandler) addLGTMIfCommented(obj *github.MungeObject, comments []*gi
 	// Assumption: The comments should be sorted (by default from github api) from oldest to latest
 	for i := len(comments) - 1; i >= 0; i-- {
 		comment := comments[i]
-		fields := strings.Fields(*comment.Body)
+		if !mungerutil.IsValidUser(comment.User) {
+			continue
+		}
+
+		if !(mungerutil.IsMungeBot(comment.User) || isReviewer(comment.User, reviewers)) {
+			continue
+		}
+
+		fields := getFields(*comment.Body)
 		if isCancelComment(fields) {
-			// "/lgtm cancell" if commented more recently than "/lgtm"
+			// "/lgtm cancel" if commented more recently than "/lgtm"
 			return
 		}
 		if isLGTMComment(fields) {
```

Comment:
> Should we invert this conditional and return early? (nit)

I tried inverting the conditional, but it became a little harder for me to read with the early return before it. It might just be me, so, if you feel strongly about it, I'll change it. 

```
for ... {
    ...
    if isLGTMComment(fields) {
	  return
    }

    if !isCancelComment(fields) {
	  continue
    }

    glog.Infof("Removing lgtm label. Reviewer (%s) cancelled", *comment.User.Login)
    obj.RemoveLabel(lgtmLabel)
    return
}
```

> Should we then check the push time vs comment time before even bothering to add lgtm? Thus fixing the race a second way?

I think that needs more thought because in case we don't honor a `/lgtm` command, we need to post a comment to that effect as well, along with the reason. Perhaps we should do the same thing as the lgtm-after-commit handler in that case, and post that message. I can follow up in a later PR if this causes issues.